### PR TITLE
fix ex11+12 and related functions

### PR DESCRIPTION
**Pull request by tpensyl**
Got examples 11 and 12 working

This involved a few updates to the main file as well to avoid errors

### CODE REVIEW

File: Examples/Gdip.Tutorial.11-Using.BRAs.to.perform.animations.ahk [hunk 1]
```diff
@@ -97,15 +97,16 @@ If (pBitmap < 1)
...
 WinHeight := Round(WinWidth//4)
 
 ; Get the width and height of a single bitmap. This bitmap is the fish banner
 Width := Gdip_GetImageWidth(pBitmap), Height := Gdip_GetImageHeight(pBitmap)
 
 ; Calculate the dimensions the fish banner will be resized to, to fit nicely on the gui
 NewWidth := 0.9*WinWidth
-NewHeight := Round((NewWidth/Width)*Height)
+NewHeight := (NewWidth/Width)*Height
+NewMargin := (WinWidth-NewWidth)/2
```
Comment (tpensyl) [inline Examples/Gdip.Tutorial.11-Using.BRAs.to.perform.animations.ahk:109]: all the functions here seem fine with floats, so I assume no need to round (already NewWidth was not an integer)

File: Gdip_All.ahk [hunk 2]
```diff
@@ -2383,7 +2383,7 @@ Gdip_TextToGraphics(pGraphics, Text, Options, Font:="Arial", Width:="", Height:=
 	Gdip_DeleteFont(hFont)
 	Gdip_DeleteFontFamily(hFamily)
 
-	return _E ? _E : ReturnRC
+	return ReturnRC
```
Comment (tpensyl) [inline Gdip_All.ahk:2386]: previously it complained about _E being undefined here (since L2375 branch was not reached)

File: Gdip_All.ahk [hunk 3]
```diff
@@ -728,7 +728,7 @@ Gdip_LibrarySubVersion()
 ;						-3 = The BRA has information missing
 ;						-4 = Could not find file inside the BRA
 
-Gdip_BitmapFromBRA(&BRAFromMemIn, File, Alternate := 0) {
+Gdip_BitmapFromBRA(BRAFromMemIn, File, Alternate := 0) {
```
Comment (tpensyl) [inline Gdip_All.ahk:731]: This now uses a Buffer, which is passed without a pointer

File: Gdip_All.ahk [hunk 4]
```diff
@@ -2812,7 +2812,7 @@ Gdip_PixelateBitmap(pBitmap, &pBitmapOut, BlockSize)
 		nCount := StrLen(MCode_PixelateBitmap)//2
 		loop nCount
 			NumPut("UChar", "0x" SubStr(MCode_PixelateBitmap, (2*A_Index)-1, 2), PixelateBitmap, A_Index-1)
-		DllCall("VirtualProtect", "UPtr", PixelateBitmap.Ptr, "UPtr", PixelateBitmap, "UInt", 0x40, "UPtr*", 0)
```
Comment (tpensyl) [inline Gdip_All.ahk:2815]: according to here, the third parameter should be the size of the target region
https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualprotect

I still have no idea what line 2831 does (what does it mean to make a dllcall with a buffer as the call name?), but the example definitely looks functional with an animated varying pixelation effect

**Review by buliasz — APPROVED** [review]
Everything LGTM, thanks!

File: Gdip_All.ahk (hunk 4 above)
Comment (buliasz) [reply Gdip_All.ahk:2815]: The 2831 call uses the option to use a pointer to a function to call instead of name. Look at https://www.autohotkey.com/docs/v2/lib/DllCall.htm: "This parameter may also consist solely of an integer, which is interpreted as the address of the function to call. Sources of such addresses include [COM](https://www.autohotkey.com/docs/v2/lib/DllCall.htm#COM) and [CallbackCreate](https://www.autohotkey.com/docs/v2/lib/CallbackCreate.htm)."
So in other words this one calls the MCode that you can see in the MCode_PixelateBitmap assignment.

Comment (tpensyl) [reply Gdip_All.ahk:2815]: oh that's wild, I didn't realize those chunks were actually code. Makes sense now, thanks!
